streaming_providers/realdebrid/utils.py

```
@@ -38,85 +42,101 @@ def get_video_url_from_realdebrid(
-        pass  # No action needed, proceed to create download link
-    else:
-        # "waiting_files_selection", "magnet_conversion", "compressing", "uploading"
-        rd_client.wait_for_status(
-            torrent_id, "waiting_files_selection", max_retries, retry_interval
+            try:
+                await rd_client.start_torrent_download(torrent_id)
+            except ProviderException as error:
+                await rd_client.delete_torrent(torrent_id)
+                raise ProviderException(
+                    f"Failed to start torrent download, {error}", "transfer_error.mp4"
+                )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)